### ISSUE TITLE
MatrixNormal complains about 1x1 matrix even when multidimensional means are provided

### ISSUE THREAD

**Issue by Spaak** [pymc-devs/pymc#4845]
The following worked fine in v3:

```python
n = 3
with Model():
    matrix_normal = MatrixNormal(
        "mat_normal",
        mu=np.random.normal(size=n),
        rowcov=np.eye(n),
        colchol=np.linalg.cholesky(np.eye(n)),
        size=(n, n),
    )
```

but now gives the following:

```
Traceback (most recent call last):
  File "/home/eelke/python/playground_pymc3_v4_strings.py", line 260, in <module>
    matrix_normal = MatrixNormal(
  File "/home/eelke/repos/pymc3/pymc3/distributions/distribution.py", line 208, in __new__
    rv_out = cls.dist(*args, rng=rng, initval=None, **kwargs)
  File "/home/eelke/repos/pymc3/pymc3/distributions/multivariate.py", line 1661, in dist
    raise ValueError(
ValueError: 1x1 Matrix was provided. Please use Normal distribution for such cases.
```

I ran into this when updating the tests in `test_distributions.py::TestStrAndLatexRepr`. I think either the check here is wrong: https://github.com/pymc-devs/pymc3/blob/7e464e59bcb0adb28df94f379b3e8d4af12bd4d1/pymc3/distributions/multivariate.py#L1656-L1659

or perhaps the input shape of `mu` is more constrained now in v4 than it was in v3 (when a 1d vector was fine). In any case the check for `mu.ndim == 1` is not sufficient to conclude that a 1x1 matrix was passed; that can also hold for a 1d vector.

## Versions and main components

* PyMC3 Version: `main` (at a061106c4555837de64c197d7217dfadfa4bfbbe)
* Aesara/Theano Version: aesara-devs/aesara@89cfea013a6963ef28c5f18640d2972bda59e3ae (HEAD, tag: rel-2.0.12)
* Python Version: 3.8.2
* Operating system: Ubuntu 20.04.2 via WSL2 (Win10)
* How did you install PyMC3: git

**Comment by ricardoV94** [comment]
@kc611 did the refactoring for V4, any insights?

**Comment by kc611** [comment]
Ah yes, the check's definitely wrong in such cases. I also tried removing it, doesn't work, seems like internally `MatrixNormal` doesn't like being given dimensions other than `(m x n)` types. 

If I'm not wrong in case of an array of means we only allow arrays broadcastable to `m x n` shape, No ? If thats the case we will simply have to broadcast `mu` in `dist` if possible else raise. And the `1x1` check should be for combination of `rowcov`'s and `colcov`s instead of `mu` since that's a better descriptor of overall shape than `mu`.

TLDR: Working on a fix...
